lib/src/state/design.dart

```diff
@@ -31,9 +32,15 @@ part 'design.g.dart';
 abstract class Design with UnusedFields implements Built<Design, DesignBuilder>, JSONSerializable {
   Design._();
 
-  factory Design({Grid grid = Grid.none}) => Design.from((b) => b
+  factory Design({Map<int, Helix> helices, Grid grid = Grid.none}){
```

Comment:
Pass in `Iterable<Helix>` and then convert it to a `Map<int, Helix>`. Use either the `Helix.idx` value to set up the Map, e.g.

```dart
factory Design({Iterable<Helix> helices, Grid grid = Grid.none}){
  var helices_map = {
    for (var helix in helices) helix.idx: helix
  };
  var design = Design.from((b) => b
    ..helices.replace(helices_map)
  );
  // ...
```

Typically I think it will be easier less verbose in unit testing code to pass in a list rather than a map.

Reply:
Thanks for the help, I just fixed it